core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -357,13 +359,22 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
    */
   def setLogLevel(logLevel: String) {
     val validLevels = Seq("ALL", "DEBUG", "ERROR", "FATAL", "INFO", "OFF", "TRACE", "WARN")
-    if (!validLevels.contains(logLevel)) {
+    // let's allow lowcase or mixed case too
+    if (!validLevels.contains(logLevel.toUpperCase)) {
```

Comment:
@srowen Thanks!  The `toUpperCase` is against the user-input `logLevel`, then check it against the `validLevels` right?  `validLevels` already contain a seq of upper case values. 

But i will change `toUpperCase` to `toUpperCase(Locale.ENGLISH)`
And i will remove the `resetLogLevel`.